src/attention-reporter.js

```diff
@@ -0,0 +1,226 @@
+ * @property {number} eventStartTime - a unix timestamp in miliseconds specifying the start time of the event
+ * @property {number} eventStopTime - a unix timestamp in miliseconds specifying the stop time of the event
+ * @interface
+ */
+
+/** 
+ * This web extension reports an attention event after the PageManager FIXME event is fired.
+ * See {@link UserEvent} for additional properties.
+ * @typedef {Object} AttentionEvent
+ * 
+ * @implements {UserEvent}
+ * @property {number} MaxPixelScrollDepth - The largest reported pixel value on the active page the user has scrolled.
+ * @property {number} maxRelativeScrollDepth - The largest reported proportion of the active page that has been scrolled already.
+ * @property {number} scrollHeight - The total scroll height of the page, taken from document.documentElement.scrollHeight.
+ * @interface
+ */
+
+/** 
+ * This web extension reports an audio event after the Pagemanager FIXME event is fired.
+ * See {@link UserEvent} for additional properties.
+ * @typedef {Object} AudioEvent
+ * 
+ * @implements {UserEvent}
+ * @interface
+ */
+
+/**
+ * A callback function for the page data event.
+ * @callback pageDataCallback
+ * @param {(AttentionEvent|AudioEvent)} details - FIXME.
+ */
+
+/**
+ * Options when adding a page data event listener.
+ * @typedef {Object} PageDataOptions
+ * @property {Array<string>} [matchPattern=[]] - The webpages of interest for the measurement, specified with WebExtensions match patterns.
+ * @property {boolean} [privateWindows=false] - Whether to measure pages in private windows.
+ */
+
+/**
+ * Function to start measurement when a listener is added
+ * TODO: deal with multiple listeners with different match patterns
+ * @private
+ * @param {EventCallbackFunction} listener - new listener being added
+ * @param {PageDataOptions} options - configuration for the events to be sent to this listener
+ */
+function addListener(listener, options) {
+    startMeasurement(options);
+}
+
+/**
+ * Function to end measurement when the last listener is removed
+ * @private
+ * @param {EventCallbackFunction} listener - listener that was just removed
+ */
+function removeListener(listener) {
+    if (!this.hasAnyListeners()) {
+        stopMeasurement();
+    }
+}
+
+/**
+ * The registered page navigation content script.
+ * @private
+ * @type {RegisteredContentScript|null}
+ */
+let registeredContentScript = null;
+/**
+ * Whether to notify the page data listener about private windows.
+ * @private
+ */
+let notifyAboutPrivateWindows = false;
+
+/**
+ * A function that is called when the content script sends a page data event message.
+ * @private
+ * @param {PageData} pageData - Information about the page.
+ */
+function pageDataListener(pageData) {
+    // If the page is in a private window and the module should not measure private windows,
+    // ignore the page
+    if(!notifyAboutPrivateWindows && pageData.privateWindow)
```

Comment:
I think this is fine because of operator precedence but I'm surprised eslint isn't bugging you to be explicit:
```suggestion
    if((!notifyAboutPrivateWindows) && pageData.privateWindow)
```

Reply:
fixed https://github.com/mozilla-rally/rally-study-01/pull/73/files#diff-4452bdc18db0dd2e5b0d942d0494128872ef601973af9ab4ebda116f3e017e87R128